packages/components/src/TimeInput.tsx

```diff
@@ -31,7 +31,9 @@ const TimeInput = React.forwardRef<HTMLInputElement, TimeInputProps>(
       onBlur = () => false,
     } = props;
     const [value, setValue] = useState(TimeUtils.formatTime(propsValue));
-    const [selection, setSelection] = useState<SelectionSegment | null>(null);
+    const [selection, setSelection] = useState<SelectionSegment | undefined>(
+      undefined
+    );
```

Comment:
You don't need to specify `undefined` with `useState`, there's a convenience overload when argument is omitted that returns `SelectionSegment | undefined`:

```suggestion
    const [selection, setSelection] = useState<SelectionSegment>();
```

From `@types/react`:
```
    /**
     * Returns a stateful value, and a function to update it.
     *
     * @version 16.8.0
     * @see https://reactjs.org/docs/hooks-reference.html#usestate
     */
    function useState<S>(initialState: S | (() => S)): [S, Dispatch<SetStateAction<S>>];
    // convenience overload when first argument is omitted
    /**
     * Returns a stateful value, and a function to update it.
     *
     * @version 16.8.0
     * @see https://reactjs.org/docs/hooks-reference.html#usestate
     */
    function useState<S = undefined>(): [S | undefined, Dispatch<SetStateAction<S | undefined>>];
```

Reply:
Nice